core/templates/dev/head/filters.js

```diff
@@ -196,8 +199,26 @@ oppia.filter('parameterizeRuleDescription', [
           } else {
             // The following case is for MultipleChoiceInput
             for (var i = 0; i < choices.length; i++) {
+              var stringChange = true;
```

Comment:
I think this is a little too complicated. Take a look at the approach [here](https://github.com/oppia/oppia/blob/develop/core/templates/dev/head/expressions/ExpressionInterpolationService.js#L44) (using string.replace()) -- see if it works for you?

Also, shouldn't this be applied to item-selection input as well?